### ISSUE TITLE
Show in UI if artifact wasn't uploaded

### ISSUE THREAD

**Issue by mitchhentges** [taskcluster/taskcluster-web#368]
In my task payload, I define the artifacts that I'm expecting.
However, I was just debugging an issue where an artifact wasn't actually produced in one of the builds. I made the mistake of assuming that if an artifact was shown in the "Artifacts" section, it's because it was successfully created and located.
However, upon further debugging, I've realized that my artifact was not actually produced. (for example, go to [this task](https://taskcluster-web.netlify.com/tasks/QON0-jDdSaKHlqEW81oogA/runs/0) and click on the "public/app-focus-arm-nightly-unsigned.apk" artifact).

TL;DR: instead of only showing that an artifact doesn't exist once it's clicked on, show in the UI if an artifact is missing.

**Comment by helfi92** [comment]
From the client side perspective, we only get the artifact URL. We'd have to download each artifact in order to see if it's missing. That would affect performance and download too much content.

If there's anything to change, it would probably be something handled in the backend: https://github.com/taskcluster/docker-worker/blob/ef974e09acbf6d36c641af0468893e9355f951cc/src/lib/features/artifacts.js#L52-L70.

**Comment by mitchhentges** [comment]
True, it would be a bummer to have to download each artifact to realize if it's missing, that's probably not worth it.

I created an [RFC suggestion](https://github.com/taskcluster/taskcluster-rfcs/issues/137) - perhaps we should start considering failing builds if they don't have their artifacts :thinking:

**Comment by helfi92** [comment]
Discussion continued in [Bug 1519963](https://bugzilla.mozilla.org/show_bug.cgi?id=1519963).
